x-pack/plugins/uptime/server/lib/synthetics_service/get_api_key.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest, SavedObjectsClientContract } from '../../../../../../src/core/server';
+import { EncryptedSavedObjectsPluginStart } from '../../../../encrypted_saved_objects/server';
+import { SecurityPluginStart } from '../../../../security/server';
+import {
+  getSyntheticsServiceAPIKey,
+  setSyntheticsServiceApiKey,
+  syntheticsServiceApiKey,
+} from '../saved_objects/service_api_key';
+
+export const getAPIKeyForSyntheticsService = async ({
+  encryptedSavedObject,
+  savedObjectsClient,
+  request,
+  security,
+}: {
+  encryptedSavedObject: EncryptedSavedObjectsPluginStart;
+  request: KibanaRequest;
+  security: SecurityPluginStart;
+  savedObjectsClient: SavedObjectsClientContract;
+}) => {
+  const encryptedClient = encryptedSavedObject.getClient({
+    includedHiddenTypes: [syntheticsServiceApiKey.name],
+  });
+
+  const apiKey = await getSyntheticsServiceAPIKey(encryptedClient);
+  if (apiKey) {
+    return apiKey;
+  }
+  return generateAndSaveAPIKey({ request, security, savedObjectsClient });
+};
+
+export const generateAndSaveAPIKey = async ({
+  security,
+  request,
+  savedObjectsClient,
+}: {
+  security: SecurityPluginStart;
+  request: KibanaRequest;
+  savedObjectsClient: SavedObjectsClientContract;
+}) => {
+  try {
+    const isApiKeysEnabled = await security.authc.apiKeys?.areAPIKeysEnabled();
+
+    if (!isApiKeysEnabled) {
+      return;
```

Comment:
I don't know a lot about the types of helpers we have with our API layer, but when we return null, what is response that is returned to the client? Status 500?

We may need to have a more specific error message here, so that we can let the user know how to remedy this situation. I don't suppose anyone wouldn't be using tls in production, which is required to use API keys, but perhaps if they are just trying to get started and testing out synthetics service, we could populate an error message to let them know to run with tls.

At any rate, this isn't necessary right now for an incremental step, but it could be good to keep in mind.

Reply:
i think in the route if the key is undefined or an error, that should be passed on to the UI. i have updated this line to bit better